graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.migrations;
+
+import com.github.zafarkhaja.semver.Version;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.cluster.State;
+import org.graylog2.indexer.ElasticsearchException;
+import org.graylog2.indexer.IndexSet;
+import org.graylog2.indexer.MongoIndexSet;
+import org.graylog2.indexer.cluster.Node;
+import org.graylog2.indexer.cluster.jest.JestUtils;
+import org.graylog2.indexer.indexset.IndexSetService;
+import org.graylog2.indexer.indices.Indices;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.graylog2.indexer.gson.GsonUtils.asJsonObject;
+import static org.graylog2.indexer.gson.GsonUtils.asString;
+
+public class V20170607164210_MigrateReopenedIndicesToAliases extends Migration {
+    private static final String REOPENED_INDEX_SETTING = "graylog2_reopened";
+
+    private final Version elasticsearchVersion;
+    private final IndexSetService indexSetService;
+    private final MongoIndexSet.Factory mongoIndexSetFactory;
+    private final Indices indices;
+    private final JestClient jestClient;
+
+    @Inject
+    public V20170607164210_MigrateReopenedIndicesToAliases(Node node,
+                                                           IndexSetService indexSetService,
+                                                           MongoIndexSet.Factory mongoIndexSetFactory,
+                                                           Indices indices,
+                                                           JestClient jestClient) {
+        this.elasticsearchVersion = node.getVersion().orElseThrow(() -> new ElasticsearchException("Unable to retrieve Elasticsearch version."));
```

Comment:
Personally I would always avoid IO during injection unless absolutely necessary as in general it is difficult to reason about how often it might be performed. Could be that in this case it is known, but IMHO it is an anti-pattern.
I strongly prefer to pass the version through the call chain, or even putting it into a mutable field in this case, maybe with a small comment explaining why that's ok in this class.